packages/clerk-js/src/ui/hooks/useInView.ts

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useRef, useState } from 'react';
+
+import type { IntersectionOptions } from '../components/OrganizationSwitcher/OtherOrganizationActions';
+
+export const useInView = (params: IntersectionOptions) => {
+  const [inView, setInView] = useState(false);
+  const observerRef = useRef<IntersectionObserver | null>(null);
+  const thresholds = Array.isArray(params.threshold) ? params.threshold : [params.threshold || 0];
+  const internalOnChange = React.useRef<IntersectionOptions['onChange']>();
+
+  internalOnChange.current = params.onChange;
+
+  const ref = useCallback((element: HTMLElement | null) => {
+    if (!element) {
+      if (observerRef.current) {
+        observerRef.current.disconnect();
+      }
+      return;
+    }
```

Comment:
Nice use of the callback ref behavior here to handle cleanup 👍

```suggestion
    // Callback refs are called with null to clear the value, so we rely on that to cleanup the observer. (ref: https://react.dev/learn/manipulating-the-dom-with-refs#how-to-manage-a-list-of-refs-using-a-ref-callback)
    if (!element) {
      if (observerRef.current) {
        observerRef.current.disconnect();
      }
      return;
    }
```

Reply:
❤️ Thanks for going the extra mile and writing the comment